pkg/distro/rhel9/images.go

```diff
@@ -698,3 +708,25 @@ func initialSetupKickstart() *fsnode.File {
 	}
 	return file
 }
+
+// fixes: https://github.com/osbuild/images/issues/352
+// Creates a unit file that create mountpoints if it doesnot exits.
+// This ensures that the filesystem created using image builder continues to work after upgrade.
+func createFilesystemMounpointService(customizations *blueprint.Customizations) *fsnode.File {
+	mntpnts := customizations.GetFilesystems()
+	if mntpnts != nil {
+		customServiceUnit := "[Unit]\nDescription=Create mountpoints if does not exists\nDefaultDependencies=no\n"
```

Comment:
It's also a best practice to include information about the *source* of systemd units.  I'd inject a comment into the generated unit at the top, like:
```suggestion
	customServiceUnit := "# Generated by osbuild/images\n[Unit]\nDescription=Create mountpoints if does not exists\nDefaultDependencies=no\n"
```

Further consider using e.g. `Documentation=https://github.com/osbuild/images` maybe?

Reply:
Added them